File: src/mmmlpmsw/comp_math/lab3/Main.java
package mmmlpmsw.comp_math.lab3;

import mmmlpmsw.comp_math.lab3.gui.UserInterface;
import mmmlpmsw.comp_math.lab3.Function;

public class Main {

    public static void main(String[] args) {

        UserInterface userInterface = new UserInterface(new Function() {
            @Override
            public double getValue(double arg) {
                return 0;
            }

            @Override
            public double getValue(double x, double y) {
                return -2 * x;
            }
        });
        userInterface.draw(1500, 700);
    }
}

File: src/main/kotlin/mmmlpmsw/comp_math/lab3/Main.kt
package mmmlpmsw.comp_math.lab3

import mmmlpmsw.comp_math.lab3.gui.UserInterface
import mmmlpmsw.comp_math.lab3.Function

object Main {

    @JvmStatic
    fun main(args: Array<String>) {
        val userInterface = UserInterface(object : Function {
            override fun getValue(arg: Double): Double {
                return 0.0
            }

            override fun getValue(x: Double, y: Double): Double {
                return -2 * x
            }
        })
        userInterface.draw(1500, 700)
    }
}